AnimatedComponents/src/components/buttons/icon-transition-button/IconTransitionButton.tsx

```diff
@@ -37,32 +63,52 @@ const IconTransitionButton = (props: IIconsTransitionButton) => {
     ],
   }))
 
+  const labelAnimatedStyles = useAnimatedStyle(() => ({
+    opacity: labelOpacity.value,
+  }))
+
   const handlePress = async () => {
     const maxTranslationX = buttonWidth.current - 50
     translateX.value = Math.min(maxTranslationX)
     rotationValue.value = withTiming(360, { duration: 1500 })
-    const halfway = 700
+    const halfway = 750
+
+    labelOpacity.value = withTiming(0, { duration: halfway })
 
     const onPressResult = await onPress()
     if (!onPressResult) {
       setTimeout(() => {
         setIconName(failedIcon)
+        setLabel(failedLabel)
+        labelOpacity.value = withTiming(1, { duration: 500 })
```

Comment:
make constants for all duration values, so it make sense that why we using that much duration.